C.js

```diff
@@ -1,77 +1,266 @@
+    
+    task.appendChild(checkboxAndText);
+    task.appendChild(editAndDeleteButtons);
+
+    return task;
+};
+
+const specifyTaskList = (task, taskElement) => {
+    const checkboxElement = taskElement.children[0].children[0];
+    if (task.list === "todo") {
+        checkboxElement.checked = false;
+        todoList.appendChild(taskElement);
+    } else {
+        checkboxElement.checked = true;
+        doneList.appendChild(taskElement);
+    }
+};
+
+const createTaskElementFromLocalStorage = (task) => {
+    const taskElement = connectTaskContent(task.id, task.description);
+    specifyTaskList(task, taskElement);    
+};
+
+const initializeTasksFromLocalStorage = () => {
+
+    tasks.forEach(task => {
+        createTaskElementFromLocalStorage(task);
+    });
+};
+
+initializeTasksFromLocalStorage();
 
 const clearTaskInput = (event) => event.target.value = "";
 
 const newTask = (event) => {
-    const task = createNewTaskElement();
-    const checkbox = createNewCheckboxElement();
-    const taskDescription = createTaskDescription(event);
-    const deleteTaskButton = createDeleteTaskButtonElement();
-    
-    task.appendChild(checkbox);
-    task.appendChild(taskDescription);
-    task.appendChild(deleteTaskButton);
+    const task = connectTaskContent(generateTaskId(taskNumber), event.target.value);
     
     todoList.appendChild(task);
+    addTaskToLocalStorage(task);
+    updateTaskNumber();
     clearTaskInput(event);
 };
 
-const switchTaskList = (event) => {
-    const task = event.target.parentElement;
-    event.target.checked ? moveToDone(task) : moveToTodo(task);
-};
-
 const moveToDone = (task) => {
     todoList.removeChild(task);
     doneList.appendChild(task);
-    task.classList.remove("toDoTask");
-    task.classList.add("doneTask");
 };
 
 const moveToTodo = (task) => {
     doneList.removeChild(task);
     todoList.appendChild(task);
-    task.classList.remove("doneTask");
-    task.classList.add("toDoTask");
 };
 
-const deleteTask = (event) => event.target.parentElement.remove();
+const addTaskToLocalStorage = (task) => {
+    const savedTask = {
+        id: task.id,
+        description: task.children[0].children[1].value,
+        list: task.children[0].children[0].checked ? "done" : "todo"
+    }
+
+    tasks.push(savedTask);
+
+    updateTasksInLocalStorage(tasks);
+};
+
+const removeTaskFromLocalStorage = (event) => {
+    const taskId = event.target.parentElement.parentElement.id;
+
+    tasks = tasks.filter(task => task.id !== taskId);
+
+    updateTasksInLocalStorage(tasks);
+};
+
+const updateTaskDescriptionInLocalStorage = (event) => {
+    const taskId = event.target.parentElement.parentElement.id;
+
+    tasks.map(task => {
+        if (task.id === taskId) {
+            return task.description = event.target.value;
+        }
+
+        return task;
+    });
+
+    updateTasksInLocalStorage(tasks);
+};
+
+const updateTaskListInLocalStorage = (event) => {

```

Comment:
consider renaming this
```suggestion
const updateTaskStatusInLocalStorage = (event) => {
```